packages/bpk-component-navigation-bar/src/BpkNavigationBar.tsx

```diff
@@ -60,49 +66,62 @@ const BpkNavigationBar = (props: Props) => {
     typeof title === 'string' ? `${id}-bpk-navigation-bar-title` : id;
 
   return (
-    // $FlowFixMe[cannot-spread-inexact] - inexact rest. See 'decisions/flowfixme.md'.
     <nav
       aria-labelledby={titleId}
       className={getClassNames(
         'bpk-navigation-bar',
+        `bpk-navigation-bar--${barStyle}`,
         sticky && 'bpk-navigation-bar__sticky',
         className,
       )}
       {...rest}
     >
       {leadingButton &&
-        cloneWithClass(leadingButton, 'bpk-navigation-bar__leading-item')}
+        cloneWithClasses(
+          leadingButton,
+          'bpk-navigation-bar__leading-item',
+          `bpk-navigation-bar__leading-item--${barStyle}`,
+        )}
       {typeof title === 'string' ? (
         <BpkText
           id={titleId}
           textStyle={TEXT_STYLES.heading5}
-          className={getClassNames('bpk-navigation-bar__title')}
+          className={getClassNames(
+            'bpk-navigation-bar__title',
+            `bpk-navigation-bar__title--${barStyle}`,
+          )}
         >
           {title}
         </BpkText>
       ) : (
         title
       )}
       {trailingButton &&
-        cloneWithClass(trailingButton, 'bpk-navigation-bar__trailing-item')}
+        cloneWithClasses(
+          trailingButton,
+          'bpk-navigation-bar__trailing-item',
+          `bpk-navigation-bar__trailing-item--${barStyle}`,
+        )}
     </nav>
   );
 };
 
 BpkNavigationBar.propTypes = {
   id: PropTypes.string.isRequired,
-  title: PropTypes.node.isRequired,
+  title: PropTypes.oneOf([PropTypes.node, PropTypes.string]).isRequired,
   className: PropTypes.string,
   leadingButton: PropTypes.element,
   trailingButton: PropTypes.element,
   sticky: PropTypes.bool,
+  barStyle: PropTypes.oneOf(Object.values(BAR_STYLES)),
 };
 
 BpkNavigationBar.defaultProps = {
```

Comment:
We can set the default props directly when destructuring the props rather than using this. We should also be getting rid of the propTypes now we're using TS, but since we export them it will be a breaking change so maybe I wouldn't add it to this PR. We do have an incoming breaking change though so maybe we can bundle it in the same release with that change in a separate PR?

Reply:
Added the default values to the destructuring expect the nulls